packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel/method_channel_load_bundle_task.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:cloud_firestore_platform_interface/cloud_firestore_platform_interface.dart';
+
+import 'method_channel_firestore.dart';
+
+import 'utils/exception.dart';
+
+class MethodChannelLoadBundleTask extends LoadBundleTaskPlatform {
+  MethodChannelLoadBundleTask(this._task, this._bundle, this._firestore)
+      : super() {
+    _controller = StreamController<LoadBundleTaskSnapshotPlatform>.broadcast(
+        onCancel: () {
+      nativePlatformStream?.cancel();
+    });
+
+    _task.then((observerId) {
+      StreamSubscription<dynamic>? nativePlatformStream;
+      nativePlatformStream =
+          MethodChannelFirebaseFirestore.loadBundleChannel(observerId!)
+              .receiveBroadcastStream(<String, Object>{
+        'bundle': _bundle,
+        'firestore': _firestore
+      }).listen((snapshot) {
+        _controller.add(LoadBundleTaskSnapshotPlatform(
+            convertToTaskState(snapshot['taskState']),
+            Map<String, dynamic>.from(snapshot)));
+
+        if (snapshot['taskState'] == 'success') {
+          _controller.close();
+          nativePlatformStream?.cancel();
+        }
+      }, onError: (error, stack) {
+        _controller.addError(convertPlatformException(error), stack);
+        _controller.close();
+        nativePlatformStream?.cancel();
+      });
+    });
+  }
+
+  final MethodChannelFirebaseFirestore _firestore;
+  final Uint8List _bundle;
+  final Future<String?> _task;
```

Comment:
This doesn't need a property (nor does _bundle and _firestore), as they are just used by the consturctor.